src/OrchardCore/OrchardCore.Users.Core/UserManagerExtensions.cs

```diff
@@ -68,4 +68,14 @@ public static async Task<bool> UpdateUserPropertiesAsync(this UserManager<IUser>
 
         return userNeedUpdate;
     }
+
+    public static IEnumerable<IUser> GetUsersNotInRolesAsync(this UserManager<IUser> userManager)
+    {
+        ArgumentNullException.ThrowIfNull(userManager);
+
+        return userManager.Users
```

Comment:
Are you sure this works? `UserManager<T>.Users` requires that the user store implement `IQueryableUserStore<T>`, which isn't the case for the YesSql store. I'd expect that extension to just throw instead of ever returning a useful result 😄

Reply:
Oh, never mind, it depends on https://github.com/OrchardCMS/OrchardCore/pull/17757 😄

Reply:
Ya, it's 😄